### PR TITLE
Fix immediate slack re-apply issue

### PR DESCRIPTION
**Pull request by storozhukBM**
This issue is reported here #119
I was able to reproduce and fix it.

### CODE REVIEW

**Comment by storozhukBM** [comment]
@rabbbit  please take a look

**Comment by rabbbit** [comment]
I'll take a look on Monday.

I'm not a fan of the test, both sleeping and asserting that time passed.

I'm also considering reverting to the basic locking version. This is the second bug we found in this one, so it's clearly beyond both of us to write this confidently.

**Comment by spencercjh** [comment]
This bug causes the failure of the rate limiter when it's a single instance of an online service. For example, a client with intermittent requests to call another service. The client holds a rate limiter as a field. The target service sometimes returns 429 (too many requests). It took me a lot of time to realize that it was a problem in this PR, and I always thought it was something in the way I invoked `Take`. 😅 

PLZ move forward with merging this PR into the master branch ASAP. 🙏 🙏 🙏 

Associated issues: #106, #119.

File: limiter_atomic_int64.go [hunk 1]
```diff
@@ -69,7 +69,7 @@ func (t *atomicInt64Limiter) Take() time.Time {
 		case timeOfNextPermissionIssue == 0 || (t.maxSlack == 0 && now-timeOfNextPermissionIssue > int64(t.perRequest)):
 			// if this is our first call or t.maxSlack == 0 we need to shrink issue time to now
 			newTimeOfNextPermissionIssue = now
-		case t.maxSlack > 0 && now-timeOfNextPermissionIssue > int64(t.maxSlack):
+		case t.maxSlack > 0 && now-timeOfNextPermissionIssue > int64(t.maxSlack)+int64(t.perRequest):
```
Comment (VovkoO) [inline limiter_atomic_int64.go:72]: Wouldn't this solution make the withSlack and withoutSlack options work exactly the same.
Looks like without slack you condition looks like:
`now - prev > period`
and withSlack
`now - prev - slack > slack + period`
Which is the same thing

Comment (storozhukBM) [reply limiter_atomic_int64.go:72]: We apply `now - int64(t.maxSlack)` only under condition and that makes the difference.

**Comment by rabbbit** [comment]
Sorry for the delay @storozhukBM - I believe I understand your fix, but propose we do something like https://github.com/uber-go/ratelimit/pull/124 to get the test in-line.

I remember about your clock implementation and I dislike where we are, but I'm not willing to give up on tests totally just yet :)

**Comment by rabbbit** [comment]
Fixed via #124 instead - thank you for the debugging/fix here @storozhukBM.

**Pull request closed without merging**